org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/ResolveSourceMappingHandlerTest.java

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Microsoft Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.jdt.ls.core.internal.handlers;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Arrays;
+
+import org.eclipse.jdt.ls.core.internal.managers.AbstractProjectsManagerBasedTest;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ResolveSourceMappingHandlerTest extends AbstractProjectsManagerBasedTest {
+
+    @Before
+	public void setup() throws Exception {
+		importProjects("maven/quickstart2");
+    }
+    
+    @Test
+    public void testResolveSourceUri() {
+        String uri = ResolveSourceMappingHandler.resolveSourceUri("at quickstart.AppTest.shouldAnswerWithTrue(AppTest.java:10)", Arrays.asList("quickstart2"));
+        assertTrue(uri.startsWith("file://"));
+        assertTrue(uri.contains("quickstart2/src/test/java/quickstart/AppTest.java"));
+    }
+
+    @Test
+    public void testResolveDependencyUri() {
+        String uri = ResolveSourceMappingHandler.resolveSourceUri("at org.junit.Assert.assertEquals(Assert.java:117)", Arrays.asList("quickstart2"));
```

Comment:
So, you have "Assert.java:117" in the project quickstart2, and while you locate the junit-4.13.jar, you're resolving to Assert.class. Shouldn't there have been a point where you looked up the source path entry, to get the actual location of Assert.java ?